// vuestic-ui/dist/es/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.js
import { defineComponent, computed, resolveComponent, openBlock, createElementBlock, normalizeClass, withKeys, withModifiers, createElementVNode, renderSlot, normalizeProps, guardReactiveProps, createVNode, createCommentVNode, createTextVNode, toDisplayString, withDirectives, Fragment, renderList, createBlock, createSlots, withCtx, vShow } from "vue";
import { u as useBem } from "../../../../composables/useBem.js";






import { u as useStrictInject } from "../../../../composables/useStrictInject.js";














import { V as VaCheckbox } from "../../../va-checkbox/index.js";















import { V as VaIcon } from "../../../va-icon/index.js";






























import { T as TreeViewKey } from "../../types.js";
import { _ as _export_sfc } from "../../../../../plugin-vue_export-helper.js";
var VaTreeNode_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ':root{--va-tree-node-contents-gap: 8px;--va-tree-node-padding: 6px;--va-tree-node-border-radius: 4px;--va-tree-node-indent: 28px;--va-tree-node-content-item-flex: 0 0 var(--va-tree-node-indent);--va-tree-node-content-body-item-flex: 1;--va-tree-node-interactive-bg-opacity: .1;--va-tree-node-children-background: linear-gradient(#adb3b9 33%, rgba(255, 255, 255, 0) 0%) 15px/1px 3px repeat-y transparent}.va-tree-node-root{display:flex;padding:var(--va-tree-node-padding);position:relative}.va-tree-node-root:before{content:"";background-color:var(--va-primary);border-radius:var(--va-tree-node-border-radius);bottom:0;left:0;opacity:0;pointer-events:none;position:absolute;right:0;top:0}.va-tree-node-root:hover:before{opacity:var(--va-tree-node-interactive-bg-opacity)}.va-tree-node-content{display:flex;flex-wrap:nowrap;align-items:center;width:100%}.va-tree-node-content__item{flex:var(--va-tree-node-content-item-flex);min-width:var(--va-tree-node-indent);line-height:1}.va-tree-node-content__item--leaf{cursor:pointer}.va-tree-node-content__body{flex:var(--va-tree-node-content-body-item-flex);width:100%}.va-tree-node-content--indent{margin-left:var(--va-tree-node-indent)}.va-tree-node-children{display:none;background:var(--va-tree-node-children-background);padding-left:var(--va-tree-node-indent);width:100%}.va-tree-node-children--expanded{display:block}.va-tree-node__checkbox{--va-checkbox-input-padding: 0}.va-tree-node--disabled{cursor:default;opacity:.4;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.va-tree-node--disabled .va-tree-node-content__item--leaf{cursor:pointer;pointer-events:all}.va-tree-node:focus-visible>.va-tree-node-root{outline:2px solid var(--va-focus)!important;border-radius:2px}.va-tree-node:focus-visible>.va-tree-node-root:before{opacity:var(--va-tree-node-interactive-bg-opacity)}\n')();
const INJECTION_ERROR_MESSAGE = "The VaTreeNode component should be used in the context of VaTreeView component";
const _sfc_main = defineComponent({
  name: "VaTreeNode",
  props: {
    node: {
      type: Object,
      required: true
    }
  },
  components: { VaCheckbox, VaIcon },
  setup: (props) => {
    const {
      iconBy,
      selectable,
      colorComputed,
      getText,
      getTrackBy,
      toggleNode,
      getNodeProperty,
      toggleCheckbox
    } = useStrictInject(TreeViewKey, INJECTION_ERROR_MESSAGE);
    const labelComputed = computed(() => getText(props.node) || "");
    const isExpandedComputed = computed(() => !!props.node.expanded);
    const iconComputed = computed(() => getNodeProperty(props.node, iconBy));
    const treeNodeClassComputed = useBem("va-tree-node", () => ({
      disabled: !!props.node.disabled
    }));
    const expandedClassComputed = useBem("va-tree-node-children", () => ({
      expanded: isExpandedComputed.value
    }));
    const indentClassComputed = useBem("va-tree-node-content", () => ({
      indent: props.node.hasChildren === false
    }));
    const tabIndexComputed = computed(() => props.node.disabled ? -1 : 0);
    const handleToggleNode = (event, node) => {
      var _a;
      if (node.expanded) {
        (_a = event.target) == null ? void 0 : _a.blur();
      } else {
        toggleNode(node);
      }
    };
    return {
      selectable,
      getText,
      getTrackBy,
      toggleNode,
      handleToggleNode,
      toggleCheckbox,
      iconComputed,
      labelComputed,
      colorComputed,
      tabIndexComputed,
      indentClassComputed,
      isExpandedComputed,
      expandedClassComputed,
      treeNodeClassComputed
    };
  }
});
const _hoisted_1 = ["aria-expanded", "aria-disabled", "aria-checked", "tabindex"];
const _hoisted_2 = {
  key: 0,
  class: "va-tree-node-content__item va-tree-node-content__item--leaf"
};
const _hoisted_3 = {
  key: 2,
  class: "va-tree-node-content__item"
};
const _hoisted_4 = { class: "va-tree-node-content__body" };
const _hoisted_5 = ["aria-hidden"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_icon = resolveComponent("va-icon");
  const _component_va_checkbox = resolveComponent("va-checkbox");
  const _component_va_tree_node = resolveComponent("va-tree-node", true);
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-tree-node", _ctx.treeNodeClassComputed]),
    role: "treeitem",
    "aria-expanded": _ctx.$props.node.expanded,
    "aria-disabled": _ctx.$props.node.disabled,
    "aria-checked": !!_ctx.$props.node.checked,
    tabindex: _ctx.tabIndexComputed,
    onKeydown: [
      _cache[3] || (_cache[3] = withKeys(withModifiers(($event) => _ctx.handleToggleNode($event, _ctx.$props.node), ["stop", "prevent"]), ["right"])),
      _cache[4] || (_cache[4] = withKeys(withModifiers(($event) => _ctx.handleToggleNode($event, _ctx.$props.node), ["stop", "prevent"]), ["left"]))
    ]
  }, [
    createElementVNode("div", {
      class: "va-tree-node-root",
      onClick: _cache[2] || (_cache[2] = withModifiers(($event) => _ctx.toggleNode(_ctx.$props.node), ["stop"]))
    }, [
      createElementVNode("div", {
        class: normalizeClass(["va-tree-node-content", _ctx.indentClassComputed])
      }, [
        _ctx.$props.node.hasChildren ? (openBlock(), createElementBlock("div", _hoisted_2, [
          renderSlot(_ctx.$slots, "icon-toggle", normalizeProps(guardReactiveProps(_ctx.$props.node)), () => [
            createVNode(_component_va_icon, {
              name: _ctx.isExpandedComputed ? "keyboard_arrow_down" : "keyboard_arrow_right",
              size: "20px"
            }, null, 8, ["name"])
          ])
        ])) : createCommentVNode("", true),
        _ctx.selectable ? (openBlock(), createElementBlock("div", {
          key: 1,
          class: "va-tree-node-content__item",
          onClick: _cache[1] || (_cache[1] = withModifiers(() => {
          }, ["stop"]))
        }, [
          renderSlot(_ctx.$slots, "checkbox", normalizeProps(guardReactiveProps(_ctx.$props.node)), () => [
            createVNode(_component_va_checkbox, {
              "model-value": _ctx.$props.node.checked,
              color: _ctx.colorComputed,
              indeterminate: "",
              "onUpdate:modelValue": _cache[0] || (_cache[0] = (v) => _ctx.toggleCheckbox(_ctx.$props.node, v)),
              class: "va-tree-node__checkbox"
            }, null, 8, ["model-value", "color"])
          ])
        ])) : createCommentVNode("", true),
        _ctx.iconComputed ? (openBlock(), createElementBlock("div", _hoisted_3, [
          renderSlot(_ctx.$slots, "icon", normalizeProps(guardReactiveProps(_ctx.$props.node)), () => [
            createVNode(_component_va_icon, {
              name: _ctx.iconComputed,
              size: "small"
            }, null, 8, ["name"])
          ])
        ])) : createCommentVNode("", true),
        createElementVNode("div", _hoisted_4, [
          renderSlot(_ctx.$slots, "content", normalizeProps(guardReactiveProps(_ctx.$props.node)), () => [
            createTextVNode(toDisplayString(_ctx.labelComputed), 1)
          ])
        ])
      ], 2)
    ]),
    withDirectives(createElementVNode("div", {
      role: "group",
      "aria-hidden": !_ctx.$props.node.expanded,
      class: normalizeClass(["va-tree-node-children", _ctx.expandedClassComputed])
    }, [
      (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.$props.node.children, (childNode) => {
        return openBlock(), createBlock(_component_va_tree_node, {
          key: _ctx.getTrackBy(childNode),
          node: childNode
        }, createSlots({ _: 2 }, [
          renderList(_ctx.$slots, (_, name) => {
            return {
              name,
              fn: withCtx((slotScope) => [
                renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(slotScope)))
              ])
            };
          })
        ]), 1032, ["node"]);
      }), 128))
    ], 10, _hoisted_5), [
      [vShow, _ctx.$props.node.hasChildren]
    ])
  ], 42, _hoisted_1);
}
var _VaTreeNode = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaTreeNode as _ };

import './VaTreeNode.css';
