// vuestic-ui/dist/es/src/components/va-card/VaCard.js
import { defineComponent, computed, openBlock, createBlock, resolveDynamicComponent, normalizeClass, normalizeStyle, withCtx, createElementBlock, createCommentVNode, createElementVNode, renderSlot } from "vue";
import { p as pick_1 } from "../../../vendor.js";
import { g as getGradientBackground } from "../../services/color/utils.js";
import { u as useBem } from "../../composables/useBem.js";
import { u as useColors } from "../../composables/useColors.js";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.js";




import { u as useRouterLinkProps, a as useRouterLink } from "../../composables/useRouterLink.js";
import { u as useTextColor } from "../../composables/useTextColor.js";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
var VaCard_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ':root{--va-card-display: block;--va-card-position: relative;--va-card-overflow: visible;--va-card-box-shadow: var(--va-box-shadow);--va-card-border-radius: .375rem;--va-card-color: #34495e;--va-card-background-color: var(--va-background-secondary);--va-card-padding: 1.25rem;--va-card-outlined-border: 3px solid var(--va-background-element);--va-card-outlined-box-shadow: none;--va-card-stripe-border-size: var(--va-stripe-border-size);--va-card-dark-color: #ffffff;--va-card-dark-background-color: #34495e;--va-card-actions-btn-margin: 4px}.va-card{display:var(--va-card-display);position:var(--va-card-position);overflow:var(--va-card-overflow);box-shadow:var(--va-card-box-shadow, var(--va-block-box-shadow));border-radius:var(--va-card-border-radius, var(--va-block-border-radius));color:var(--va-card-color);background-color:var(--va-card-background-color);font-family:var(--va-font-family)}.va-card__inner{width:100%;height:100%}.va-card__inner>div:first-child{border-top-right-radius:var(--va-card-border-radius);border-top-left-radius:var(--va-card-border-radius)}.va-card__inner>div:last-child{border-bottom-right-radius:var(--va-card-border-radius);border-bottom-left-radius:var(--va-card-border-radius)}.va-card--square{border-radius:0}.va-card--outlined{box-shadow:var(--va-card-outlined-box-shadow);border:var(--va-card-outlined-border, var(--va-block-border))}.va-card--no-border{border:none}.va-card--disabled{cursor:default;opacity:.4;pointer-events:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.va-card--link{cursor:pointer}.va-card__stripe{content:"";position:absolute;width:100%;height:var(--va-card-stripe-border-size);top:0;left:0}.va-card__title,.va-card__content,.va-card__actions,.va-card__actions--vertical{padding:var(--va-card-padding)}.va-card__title+.va-card__title,.va-card__title+.va-card__content,.va-card__title+.va-card__actions,.va-card__title+.va-card_actions__vertical,.va-card__content+.va-card__title,.va-card__content+.va-card__content,.va-card__content+.va-card__actions,.va-card__content+.va-card_actions__vertical,.va-card__actions+.va-card__title,.va-card__actions+.va-card__content,.va-card__actions+.va-card__actions,.va-card__actions+.va-card_actions__vertical,.va-card__actions--vertical+.va-card__title,.va-card__actions--vertical+.va-card__content,.va-card__actions--vertical+.va-card__actions,.va-card__actions--vertical+.va-card_actions__vertical{padding-top:0}.va-card__title{display:flex;align-items:center;font-size:.625rem;letter-spacing:.6px;line-height:1.2;font-weight:700;text-transform:uppercase}\n')();
const _sfc_main = defineComponent({
  name: "VaCard",
  emits: ["click"],
  props: {
    ...useRouterLinkProps,
    ...useComponentPresetProp,
    tag: { type: String, default: "div" },
    square: { type: Boolean, default: false },
    outlined: { type: Boolean, default: false },
    bordered: { type: Boolean, default: true },
    disabled: { type: Boolean, default: false },
    href: { type: String, default: "" },
    target: { type: String, default: "" },
    stripe: { type: Boolean, default: false },
    stripeColor: { type: String, default: "" },
    gradient: { type: Boolean, default: false },
    textColor: { type: String },
    color: { type: String, default: "background-secondary" }
  },
  setup(props) {
    const { getColor } = useColors();
    const { isLinkTag, tagComputed, hrefComputed } = useRouterLink(props);
    const { textColorComputed } = useTextColor();
    const stripeStyles = computed(() => ({ background: getColor(props.stripeColor) }));
    const classComputed = useBem("va-card", () => ({
      ...pick_1(props, ["square", "outlined", "disabled"]),
      noBorder: !props.bordered,
      link: isLinkTag.value
    }));
    const cardStyles = computed(() => {
      const background = props.gradient && props.color ? getGradientBackground(getColor(props.color)) : getColor(props.color);
      return {
        background,
        color: textColorComputed.value
      };
    });
    return {
      classComputed,
      cardStyles,
      stripeStyles,
      tagComputed,
      hrefComputed
    };
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createBlock(resolveDynamicComponent(_ctx.tagComputed), {
    class: normalizeClass(["va-card", _ctx.classComputed]),
    style: normalizeStyle(_ctx.cardStyles),
    href: _ctx.hrefComputed,
    target: _ctx.target,
    to: _ctx.to,
    replace: _ctx.replace,
    exact: _ctx.exact,
    "active-class": _ctx.activeClass,
    "exact-active-class": _ctx.exactActiveClass
  }, {
    default: withCtx(() => [
      _ctx.stripe ? (openBlock(), createElementBlock("div", {
        key: 0,
        class: "va-card__stripe",
        style: normalizeStyle(_ctx.stripeStyles)
      }, null, 4)) : createCommentVNode("", true),
      createElementVNode("div", {
        class: "va-card__inner",
        onClick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("click", $event))
      }, [
        renderSlot(_ctx.$slots, "default")
      ])
    ]),
    _: 3
  }, 8, ["class", "style", "href", "target", "to", "replace", "exact", "active-class", "exact-active-class"]);
}
var _VaCard = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaCard as _ };

import './VaCard.css';
