// vuestic-ui/dist/es/src/components/va-list/VaListItemSection.js
import { defineComponent, computed, openBlock, createElementBlock, normalizeClass, renderSlot } from "vue";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.js";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
var VaListItemSection_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ":root{--va-list-width: 100%;--va-list-label-text-align: center;--va-list-label-padding: .3rem 0;--va-list-item-display: flex;--va-list-item-align-items: center;--va-list-item-width: 100%;--va-list-item-height: 100%;--va-list-item-label-color: #34495e;--va-list-item-label-display: -webkit-box;--va-list-item-label-box-orient: vertical;--va-list-item-label-overflow: hidden;--va-list-item-label-line-height: normal;--va-list-item-label-caption-font-size: 85%;--va-list-item-label-caption-color: #babfc2;--va-list-item-section-display: flex;--va-list-item-section-icon-min-width: 1.5rem;--va-list-item-section-icon-align-items: center;--va-list-item-section-icon-justify-content: center;--va-list-item-section-icon-margin: .6rem .75rem;--va-list-item-section-icon-font-size: 1.25rem;--va-list-item-section-avatar-min-width: 3rem}.va-list-item-section{display:flex;font-family:var(--va-font-family)}.va-list-item-section--main{min-width:0;flex-direction:column;flex:1 0}.va-list-item-section--icon{min-width:var(--va-list-item-section-icon-min-width);align-items:var(--va-list-item-section-icon-align-items);justify-content:var(--va-list-item-section-icon-justify-content);margin:var(--va-list-item-section-icon-margin)}.va-list-item-section--icon .va-icon{font-size:var(--va-list-item-section-icon-font-size)}.va-list-item-section--avatar{min-width:var(--va-list-item-section-avatar-min-width)}.va-list-item-section+.va-list-item-section{margin-left:.5rem}.va-list-item-section+.va-list-item-section--icon:last-child{margin-left:1rem}\n")();
const _sfc_main = defineComponent({
  name: "VaListItemSection",
  props: {
    ...useComponentPresetProp,
    icon: { type: Boolean, default: false },
    avatar: { type: Boolean, default: false }
  },
  setup: (props) => ({
    computedClass: computed(() => ({
      "va-list-item-section--main": !props.icon && !props.avatar,
      "va-list-item-section--icon": props.icon,
      "va-list-item-section--avatar": props.avatar
    }))
  })
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-list-item-section", _ctx.computedClass])
  }, [
    renderSlot(_ctx.$slots, "default")
  ], 2);
}
var _VaListItemSection = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaListItemSection as _ };

import './VaListItemSection.css';
