// vuestic-ui/dist/esm-node/src/components/va-sidebar/VaSidebar.mjs
import { defineComponent, ref, computed, resolveComponent, openBlock, createElementBlock, normalizeClass, normalizeStyle, createElementVNode, createVNode, withCtx, renderSlot } from "vue";
import { g as getGradientBackground } from "../../services/color/utils.mjs";
import { u as useBem } from "../../composables/useBem.mjs";
import { u as useColors } from "../../composables/useColors.mjs";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.mjs";





import { u as useTextColor } from "../../composables/useTextColor.mjs";
import { u as useSidebar } from "./hooks/useSidebar.mjs";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.mjs";
var VaSidebar_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ":root{--va-sidebar-min-height: 100%;--va-sidebar-height: 100%;--va-sidebar-position: relative;--va-sidebar-top: 0;--va-sidebar-left: 0;--va-sidebar-transition: var(--va-transition);--va-sidebar-z-index: 1;--va-sidebar-menu-max-height: 100%;--va-sidebar-menu-margin-bottom: 0;--va-sidebar-menu-list-style: none;--va-sidebar-menu-padding-left: 0;--va-sidebar-menu-overflow-y: auto;--va-sidebar-menu-overflow-x: hidden;--va-sidebar-item-active-border-size: 4px;--va-sidebar-item-transition: var(--va-transition);--va-sidebar-item-content-padding: 1rem;--va-sidebar-item-content-gap: .5rem;--va-sidebar-item-title-white-space: nowrap}.va-sidebar{min-height:var(--va-sidebar-min-height);height:var(--va-sidebar-height);position:var(--va-sidebar-position);top:var(--va-sidebar-top);left:var(--va-sidebar-left);z-index:var(--va-sidebar-z-index);font-family:var(--va-font-family);display:inline-flex}.va-sidebar__menu{display:flex;flex-direction:column;max-height:var(--va-sidebar-menu-max-height);margin-bottom:var(--va-sidebar-menu-margin-bottom);list-style:var(--va-sidebar-menu-list-style);padding-left:var(--va-sidebar-menu-padding-left);overflow-y:var(--va-sidebar-menu-overflow-y);overflow-x:var(--va-sidebar-menu-overflow-x);scrollbar-color:var(--va-primary) transparent;scrollbar-width:thin}.va-sidebar__menu::-webkit-scrollbar{width:4px;height:4px}.va-sidebar__menu::-webkit-scrollbar-track{box-shadow:none;border-radius:10px}.va-sidebar__menu::-webkit-scrollbar-thumb{background:var(--va-primary);opacity:.3;border-radius:2px}.va-sidebar--animated,.va-sidebar--animated .va-sidebar__menu{transition:var(--va-sidebar-transition)}.va-sidebar--minimized{left:0}.va-sidebar--minimized .va-sidebar__title{display:none}.va-sidebar--right{left:auto;right:0}\n")();
const _sfc_main = defineComponent({
  name: "VaSidebar",
  props: {
    ...useComponentPresetProp,
    activeColor: { type: String, default: "primary" },
    hoverColor: { type: String, default: void 0 },
    hoverOpacity: {
      type: Number,
      default: 0.2,
      validator: (v) => v >= 0 && v <= 1
    },
    borderColor: { type: String, default: void 0 },
    color: { type: String, default: "background-element" },
    textColor: { type: String },
    gradient: { type: Boolean, default: false },
    minimized: { type: Boolean, default: false },
    hoverable: { type: Boolean, default: false },
    position: {
      type: String,
      default: "left",
      validator: (v) => ["left", "right"].includes(v)
    },
    width: { type: String, default: "16rem" },
    minimizedWidth: { type: String, default: "4rem" },
    modelValue: { type: Boolean, default: true },
    animated: { type: Boolean, default: true }
  },
  setup(props) {
    const { getColor } = useColors();
    useSidebar(props);
    const isHovered = ref(false);
    const isMinimized = computed(() => props.minimized || props.hoverable && !isHovered.value);
    const computedWidth = computed(() => {
      if (!props.modelValue) {
        return 0;
      }
      return isMinimized.value ? props.minimizedWidth : props.width;
    });
    const { textColorComputed } = useTextColor();
    const computedStyle = computed(() => {
      const backgroundColor = getColor(props.color);
      const color = textColorComputed.value;
      return {
        color,
        backgroundColor,
        backgroundImage: props.gradient ? getGradientBackground(backgroundColor) : void 0
      };
    });
    const computedClass = useBem("va-sidebar", () => ({
      minimized: isMinimized.value,
      right: props.position === "right",
      animated: props.animated
    }));
    const updateHoverState = (newHoverState) => {
      isHovered.value = props.hoverable && newHoverState;
    };
    return {
      computedWidth,
      computedClass,
      computedStyle,
      updateHoverState,
      vaSidebarItemProps: computed(() => ({
        textColor: props.textColor,
        activeColor: props.activeColor,
        hoverColor: props.hoverColor,
        borderColor: props.borderColor,
        hoverOpacity: props.hoverOpacity
      }))
    };
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_config = resolveComponent("va-config");
  return openBlock(), createElementBlock("aside", {
    class: normalizeClass(["va-sidebar", _ctx.computedClass]),
    style: normalizeStyle(_ctx.computedStyle),
    onMouseenter: _cache[0] || (_cache[0] = ($event) => _ctx.updateHoverState(true)),
    onMouseleave: _cache[1] || (_cache[1] = ($event) => _ctx.updateHoverState(false))
  }, [
    createElementVNode("div", {
      class: "va-sidebar__menu",
      style: normalizeStyle(`width: ${_ctx.computedWidth};`)
    }, [
      createVNode(_component_va_config, {
        components: { VaSidebarItem: _ctx.vaSidebarItemProps }
      }, {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default")
        ]),
        _: 3
      }, 8, ["components"])
    ], 4)
  ], 38);
}
var _VaSidebar = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaSidebar as _ };
