// vuestic-ui/dist/esm-node/src/components/va-color-palette/VaColorPalette.mjs
import { defineComponent, resolveComponent, openBlock, createElementBlock, createElementVNode, Fragment, renderList, createBlock } from "vue";



import { u as useComponentPresetProp } from "../../composables/useComponentPreset.mjs";



import { u as useStatefulEmits, a as useStatefulProps, b as useStateful } from "../../composables/useStateful.mjs";
import { u as useTranslation } from "../../composables/useTranslation.mjs";
import { V as VaColorIndicator } from "../va-color-indicator/index.mjs";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.mjs";
var VaColorPalette_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ".va-color-palette{padding-top:3px}.va-color-palette__colors{display:flex}.va-color-palette__colors>*{margin-right:.25rem}.va-color-palette__colors>*:last-child{margin-right:0}\n")();
const _sfc_main = defineComponent({
  name: "VaColorPalette",
  components: { VaColorIndicator },
  emits: useStatefulEmits,
  props: {
    ...useStatefulProps,
    ...useComponentPresetProp,
    modelValue: { type: String, default: null },
    palette: { type: Array, default: () => [] },
    indicator: {
      type: String,
      default: "dot",
      validator: (value) => ["dot", "square"].includes(value)
    }
  },
  setup(props, { emit }) {
    const { valueComputed } = useStateful(props, emit);
    return {
      ...useTranslation(),
      valueComputed,
      isSelected: (color) => valueComputed.value === color
    };
  }
});
const _hoisted_1 = { class: "va-color-palette" };
const _hoisted_2 = ["aria-label"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_color_indicator = resolveComponent("va-color-indicator");
  return openBlock(), createElementBlock("div", _hoisted_1, [
    createElementVNode("ul", {
      class: "va-color-palette__colors",
      role: "listbox",
      "aria-label": _ctx.t("colorSelection")
    }, [
      (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.palette, (color, index) => {
        return openBlock(), createBlock(_component_va_color_indicator, {
          key: index,
          role: "option",
          "aria-label": _ctx.t("color", { color }),
          "aria-selected": _ctx.isSelected(color),
          tabindex: "0",
          modelValue: _ctx.isSelected(color),
          color,
          square: _ctx.indicator === "square",
          "onUpdate:modelValue": ($event) => _ctx.valueComputed = color
        }, null, 8, ["aria-label", "aria-selected", "modelValue", "color", "square", "onUpdate:modelValue"]);
      }), 128))
    ], 8, _hoisted_2)
  ]);
}
var _VaColorPalette = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaColorPalette as _ };
