// vuestic-ui/dist/esm-node/src/components/va-timeline/VaTimelineSeparator/VaTimelineSeparator.mjs
import { defineComponent, h } from "vue";

import { u as useColors } from "../../../composables/useColors.mjs";
import { u as useComponentPresetProp } from "../../../composables/useComponentPreset.mjs";





var VaTimelineSeparator_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ":root{--va-timeline-separator-display: flex;--va-timeline-separator-align-items: center;--va-timeline-separator-line-transition: background-color ease .5s;--va-timeline-separator-line-width: .125rem;--va-timeline-separator-line-height: .125rem;--va-timeline-separator-line-flex: 1;--va-timeline-separator-center-transition: background-color ease .5s;--va-timeline-separator-center-width: .5rem;--va-timeline-separator-center-height: .5rem;--va-timeline-separator-center-border-radius: .5rem;--va-timeline-separator-vertical-flex-direction: column}.va-timeline-separator{display:var(--va-timeline-separator-display);align-items:var(--va-timeline-separator-align-items)}.va-timeline-separator--vertical{flex-direction:var(--va-timeline-separator-vertical-flex-direction)}.va-timeline-separator__line{transition:var(--va-timeline-separator-line-transition);width:var(--va-timeline-separator-line-width);height:var(--va-timeline-separator-line-height);flex:var(--va-timeline-separator-line-flex)}.va-timeline-separator__center{transition:var(--va-timeline-separator-center-transition);width:var(--va-timeline-separator-center-width);height:var(--va-timeline-separator-center-height);border-radius:var(--va-timeline-separator-center-border-radius)}\n")();
const componentName = "va-timeline-separator";
const _sfc_main = defineComponent({
  name: componentName,
  props: {
    ...useComponentPresetProp,
    color: { type: String, default: "primary" },
    vertical: { type: Boolean },
    active: { type: Boolean },
    activePrevious: { type: Boolean },
    activeNext: { type: Boolean }
  },
  setup(props) {
    const { getColor } = useColors();
    return () => h("div", {
      class: {
        [componentName]: true,
        [`${componentName}--vertical`]: props.vertical
      }
    }, [
      h("div", {
        class: {
          [`${componentName}__line`]: true,
          [`${componentName}__line--active`]: props.activePrevious
        },
        style: {
          backgroundColor: getColor(props.activePrevious ? props.color : "divider")
        }
      }),
      h("div", {
        class: {
          [`${componentName}__center`]: true,
          [`${componentName}__center--active`]: props.active
        },
        style: {
          backgroundColor: getColor(props.active ? props.color : "divider")
        }
      }),
      h("div", {
        class: {
          [`${componentName}__line`]: true,
          [`${componentName}__line--active`]: props.activeNext
        },
        style: {
          backgroundColor: getColor(props.activeNext ? props.color : "divider")
        }
      })
    ]);
  }
});
export { _sfc_main as _ };
