// vuestic-ui/dist/es/src/components/va-time-picker/components/VaTimePickerColumn/VaTimePickerColumn.js
import { V as VaTimePickerColumnCell } from "../VaTimePickerColumnCell.js";
import { defineComponent, shallowRef, watch, onMounted, nextTick, computed, resolveComponent, openBlock, createElementBlock, withKeys, withModifiers, normalizeStyle, Fragment, renderList, createBlock, withCtx, createElementVNode, normalizeClass, renderSlot, normalizeProps, guardReactiveProps, createTextVNode, toDisplayString } from "vue";
import { d as debounce_1 } from "../../../../../vendor.js";
import { a as useFocusEmits, u as useFocus } from "../../../../composables/useFocus.js";



import { u as useElementBackground } from "../../../../composables/useElementBackground/useElementBackground.js";

import { u as useSyncProp } from "../../../../composables/useSyncProp.js";
import { u as useTextColor } from "../../../../composables/useTextColor.js";
import { _ as _export_sfc } from "../../../../../plugin-vue_export-helper.js";
var VaTimePickerColumn_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => '.va-time-picker-column{--va-time-picker-column-gap-height: 50px;--va-time-picker-column-border-right: 1px solid var(--va-background-element);--va-time-picker-cell-height: 30px;--va-time-picker-cell-width: 40px;--va-time-picker-cell-cursor: pointer;--va-time-picker-cell-active-background: var(--va-primary);--va-time-picker-cell-active-color: var(--va-primary);--va-time-picker-cell-background-color-hover: transparent;--va-time-picker-cell-active-background-opacity: .05;--va-time-picker-cell-active-background-opacity-hover: .1}.va-time-picker-column{overflow-y:scroll;-ms-overflow-style:none;scrollbar-width:none;height:100%;border-right:var(--va-time-picker-column-border-right)}.va-time-picker-column::-webkit-scrollbar{display:none}.va-time-picker-column:before,.va-time-picker-column:after{content:"";display:block;height:var(--va-time-picker-column-gap-height);width:100%}.va-time-picker-column:last-child{border-right:0}.va-time-picker-column .va-time-picker-cell{height:var(--va-time-picker-cell-height);line-height:var(--va-time-picker-cell-height);width:var(--va-time-picker-cell-width);text-align:center;cursor:var(--va-time-picker-cell-cursor);-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.va-time-picker-column .va-time-picker-cell--active{position:relative;color:var(--va-time-picker-cell-active-color);z-index:0}.va-time-picker-column .va-time-picker-cell--active:before{background:var(--va-time-picker-cell-active-background);position:absolute;height:100%;width:100%;left:0;right:0;content:"";z-index:-1;opacity:var(--va-time-picker-cell-active-background-opacity)}.va-time-picker-column .va-time-picker-cell:hover{background:var(--va-time-picker-cell-background-color-hover)}.va-time-picker-column:focus-visible .va-time-picker-cell--active{outline:2px solid var(--va-focus)!important;border-radius:4px;outline-offset:-2px}\n')();
const _sfc_main = defineComponent({
  name: "VaTimePickerColumn",
  components: { VaTimePickerColumnCell },
  props: {
    items: { type: Array, default: () => [] },
    activeItemIndex: { type: Number, default: 0 },
    cellHeight: { type: Number, default: 30 }
  },
  emits: ["item-selected", "update:activeItemIndex", ...useFocusEmits],
  setup(props, { emit }) {
    const rootElement = shallowRef();
    const { focus, blur } = useFocus(rootElement, emit);
    const [syncActiveItemIndex] = useSyncProp("activeItemIndex", props, emit);
    watch(syncActiveItemIndex, (newVal) => {
      scrollTo(newVal);
    });
    onMounted(() => scrollTo(syncActiveItemIndex.value, false));
    const scrollTo = (index, animated = true) => {
      nextTick(() => {
        var _a, _b;
        (_b = (_a = rootElement.value) == null ? void 0 : _a.scrollTo) == null ? void 0 : _b.call(_a, {
          behavior: animated ? "smooth" : "auto",
          top: index * props.cellHeight
        });
      });
    };
    const makeActiveByIndex = (index) => {
      syncActiveItemIndex.value = index;
      nextTick(() => scrollTo(syncActiveItemIndex.value));
    };
    const makeActiveNext = (times) => {
      syncActiveItemIndex.value = (syncActiveItemIndex.value + (times || 1)) % props.items.length;
      nextTick(() => scrollTo(syncActiveItemIndex.value));
    };
    const makeActivePrev = (times) => {
      syncActiveItemIndex.value = (syncActiveItemIndex.value - (times || 1) + props.items.length) % props.items.length;
      nextTick(() => scrollTo(syncActiveItemIndex.value));
    };
    const onCellClick = (index) => {
      syncActiveItemIndex.value = index;
    };
    const formatCell = (n) => {
      if (!Number.isInteger(n)) {
        return n;
      }
      return n < 10 ? `0${n}` : `${n}`;
    };
    const { background } = useElementBackground(rootElement);
    const { textColorComputed } = useTextColor(background);
    const styleComputed = computed(() => ({
      color: textColorComputed.value
    }));
    const getIndex = () => {
      const scrollTop = rootElement.value.scrollTop;
      const calculatedIndex = Math.max((scrollTop - scrollTop % props.cellHeight) / props.cellHeight, scrollTop / props.cellHeight);
      if (syncActiveItemIndex.value * props.cellHeight < scrollTop) {
        return Math.ceil(calculatedIndex);
      } else if (syncActiveItemIndex.value * props.cellHeight > scrollTop) {
        return Math.floor(calculatedIndex);
      } else {
        return Math.round(calculatedIndex);
      }
    };
    const onScroll = debounce_1(() => {
      if (rootElement.value && syncActiveItemIndex.value !== -1) {
        syncActiveItemIndex.value = getIndex();
      }
    }, 200);
    return {
      rootElement,
      makeActiveNext,
      makeActivePrev,
      makeActiveByIndex,
      onScroll,
      onCellClick,
      formatCell,
      styleComputed,
      focus,
      blur
    };
  }
});
const _hoisted_1 = ["onClick"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_VaTimePickerColumnCell = resolveComponent("VaTimePickerColumnCell");
  return openBlock(), createElementBlock("div", {
    ref: "rootElement",
    tabindex: "0",
    class: "va-time-picker-column",
    onKeydown: [
      _cache[0] || (_cache[0] = withKeys(withModifiers(($event) => _ctx.makeActiveNext(), ["stop", "prevent"]), ["down"])),
      _cache[1] || (_cache[1] = withKeys(withModifiers(($event) => _ctx.makeActiveNext(5), ["stop", "prevent"]), ["space"])),
      _cache[2] || (_cache[2] = withKeys(withModifiers(($event) => _ctx.makeActivePrev(), ["stop", "prevent"]), ["up"]))
    ],
    style: normalizeStyle(_ctx.styleComputed)
  }, [
    (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.items, (item, index) => {
      return openBlock(), createBlock(_component_VaTimePickerColumnCell, {
        key: item,
        onScrollPassive: _ctx.onScroll,
        onTouchmovePassive: _ctx.onScroll,
        onMousewheelPassive: _ctx.onScroll
      }, {
        default: withCtx(() => [
          createElementVNode("div", {
            class: normalizeClass(["va-time-picker-cell", { "va-time-picker-cell--active": index === _ctx.$props.activeItemIndex }]),
            onClick: ($event) => _ctx.onCellClick(index)
          }, [
            renderSlot(_ctx.$slots, "cell", normalizeProps(guardReactiveProps({ item, index, activeItemIndex: _ctx.activeItemIndex, items: _ctx.items, formattedItem: _ctx.formatCell(item) })), () => [
              createTextVNode(toDisplayString(_ctx.formatCell(item)), 1)
            ])
          ], 10, _hoisted_1)
        ]),
        _: 2
      }, 1032, ["onScrollPassive", "onTouchmovePassive", "onMousewheelPassive"]);
    }), 128))
  ], 36);
}
var _VaTimePickerColumn = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaTimePickerColumn as _ };

import './VaTimePickerColumn.css';
