// vuestic-ui/dist/es/src/components/va-avatar/VaAvatar.js
import { defineComponent, computed, resolveComponent, openBlock, createElementBlock, normalizeStyle, renderSlot, createBlock, createCommentVNode } from "vue";

import { u as useColors } from "../../composables/useColors.js";
import { u as useComponentPresetProp } from "../../composables/useComponentPreset.js";




import { u as useLoadingProps } from "../../composables/useLoading.js";

import { u as useSizeProps, a as useSize } from "../../composables/useSize.js";
import { u as useTextColor } from "../../composables/useTextColor.js";
import { V as VaIcon } from "../va-icon/index.js";
import { V as VaProgressCircle } from "../va-progress-circle/index.js";
import { _ as _export_sfc } from "../../../plugin-vue_export-helper.js";
var VaAvatar_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ":root{--va-avatar-display: inline-flex;--va-avatar-align-items: center;--va-avatar-justify-content: center;--va-avatar-text-align: center;--va-avatar-vertical-align: middle;--va-avatar-position: relative;--va-avatar-line-height: normal;--va-avatar-border-radius: 50%}.va-avatar{align-items:var(--va-avatar-align-items);display:var(--va-avatar-display);justify-content:var(--va-avatar-justify-content);line-height:var(--va-avatar-line-height);position:var(--va-avatar-position);text-align:var(--va-avatar-text-align);vertical-align:var(--va-avatar-vertical-align);border-radius:var(--va-avatar-border-radius);font-family:var(--va-font-family)}.va-avatar img,.va-avatar svg{border-radius:inherit;display:inline-flex;height:inherit;width:inherit;margin:auto}\n")();
const _sfc_main = defineComponent({
  name: "VaAvatar",
  components: { VaIcon, VaProgressCircle },
  props: {
    ...useLoadingProps,
    ...useSizeProps,
    ...useComponentPresetProp,
    color: { type: String, default: "info" },
    textColor: { type: String },
    square: { type: Boolean, default: false },
    icon: { type: String, default: "" },
    src: { type: String, default: null },
    alt: { type: String, default: "" },
    fontSize: { type: String, default: "" }
  },
  setup(props) {
    const { getColor } = useColors();
    const colorComputed = computed(() => getColor(props.color));
    const { sizeComputed, fontSizeComputed } = useSize(props, "VaAvatar");
    const { textColorComputed } = useTextColor();
    const computedStyle = computed(() => ({
      color: textColorComputed.value,
      backgroundColor: props.loading ? "transparent" : colorComputed.value,
      borderRadius: props.square ? 0 : "",
      fontSize: props.fontSize || fontSizeComputed.value,
      width: sizeComputed.value,
      minWidth: sizeComputed.value,
      height: sizeComputed.value
    }));
    return {
      sizeComputed,
      computedStyle,
      colorComputed
    };
  }
});
const _hoisted_1 = ["aria-hidden"];
const _hoisted_2 = ["src", "alt"];
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_va_progress_circle = resolveComponent("va-progress-circle");
  const _component_va_icon = resolveComponent("va-icon");
  return openBlock(), createElementBlock("div", {
    class: "va-avatar",
    style: normalizeStyle(_ctx.computedStyle),
    "aria-hidden": !_ctx.$props.src,
    "aria-live": "polite"
  }, [
    renderSlot(_ctx.$slots, "default", {}, () => [
      _ctx.$props.loading ? (openBlock(), createBlock(_component_va_progress_circle, {
        key: 0,
        size: _ctx.sizeComputed,
        color: _ctx.colorComputed,
        indeterminate: ""
      }, null, 8, ["size", "color"])) : _ctx.$props.src ? (openBlock(), createElementBlock("img", {
        key: 1,
        src: _ctx.$props.src,
        alt: _ctx.$props.alt
      }, null, 8, _hoisted_2)) : _ctx.$props.icon ? (openBlock(), createBlock(_component_va_icon, {
        key: 2,
        name: _ctx.$props.icon
      }, null, 8, ["name"])) : createCommentVNode("", true)
    ])
  ], 12, _hoisted_1);
}
var _VaAvatar = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { _VaAvatar as _ };

import './VaAvatar.css';
