// vuestic-ui/dist/esm-node/src/components/va-date-picker/components/VaDatePickerCell.mjs
import { defineComponent, computed, useCssVars, openBlock, createElementBlock, normalizeClass, withKeys, withModifiers, renderSlot } from "vue";

import { u as useColors } from "../../../composables/useColors.mjs";





import { u as useTextColor } from "../../../composables/useTextColor.mjs";
import { _ as _export_sfc } from "../../../../plugin-vue_export-helper.mjs";
var VaDatePickerCell_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => '.va-date-picker-cell{position:relative;color:var(--va-date-picker-text-color);line-height:var(--va-date-picker-cell-size);min-height:var(--va-date-picker-cell-size);min-width:var(--va-date-picker-cell-size);cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;border-radius:var(--va-date-picker-cell-radius);box-sizing:border-box;text-align:center;z-index:1}.va-date-picker-cell:after,.va-date-picker-cell:before{content:"";position:absolute;height:100%;width:100%;left:0;z-index:-1;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;box-sizing:border-box;border-radius:var(--va-date-picker-cell-radius)}.va-date-picker-cell_clear{cursor:default;opacity:0}.va-date-picker-cell_highlighted-weekend{color:var(--va-date-picker-weekends-color)}.va-date-picker-cell_in-range{color:var(--va-date-picker-color)}.va-date-picker-cell_in-range:after{background-color:var(--va-date-picker-selected-background);opacity:var(--va-date-picker-cell-background-opacity-hover)}.va-date-picker-cell_today{color:var(--va-date-picker-color);font-weight:700}.va-date-picker-cell_today:before{border:2px solid var(--va-date-picker-color)}.va-date-picker-cell_selected{background-color:var(--4287396d);color:var(--6cf74d8a)}.va-date-picker-cell_other-month{opacity:var(--va-date-picker-cell-opacity)}.va-date-picker-cell:hover:not(.va-date-picker-cell_selected):not(.va-date-picker-cell_readonly):not(.va-date-picker-cell_disabled):not(.va-date-picker-cell_other-month),.va-date-picker-cell_focused:not(.va-date-picker-cell_selected):not(.va-date-picker-cell_readonly):not(.va-date-picker-cell_disabled):not(.va-date-picker-cell_other-month),.va-date-picker-cell_focused:hover:not(.va-date-picker-cell_selected):not(.va-date-picker-cell_readonly):not(.va-date-picker-cell_disabled):not(.va-date-picker-cell_other-month){color:var(--va-date-picker-color)}.va-date-picker-cell:hover:not(.va-date-picker-cell_selected):not(.va-date-picker-cell_readonly):not(.va-date-picker-cell_disabled):not(.va-date-picker-cell_other-month):after,.va-date-picker-cell_focused:not(.va-date-picker-cell_selected):not(.va-date-picker-cell_readonly):not(.va-date-picker-cell_disabled):not(.va-date-picker-cell_other-month):after,.va-date-picker-cell_focused:hover:not(.va-date-picker-cell_selected):not(.va-date-picker-cell_readonly):not(.va-date-picker-cell_disabled):not(.va-date-picker-cell_other-month):after{background-color:var(--va-date-picker-selected-background);opacity:var(--va-date-picker-cell-background-opacity-hover)}.va-date-picker-cell_disabled{cursor:not-allowed;opacity:var(--va-date-picker-cell-opacity)}.va-date-picker-cell_disabled:not(.va-date-picker-cell_today){opacity:.5}.va-date-picker-cell_disabled:after{opacity:var(--va-date-picker-cell-opacity);border:none;height:2px;width:50%;background:currentColor;top:50%;left:50%;transform:translate(-50%,-50%)}.va-date-picker-cell_readonly{cursor:default}.va-date-picker-cell_readonly:before{display:none}\n')();
const __default__ = defineComponent({
  name: "VaDatePickerCell",
  props: {
    otherMonth: { type: Boolean, default: false },
    today: { type: Boolean, default: false },
    inRange: { type: Boolean, default: false },
    disabled: { type: Boolean, default: false },
    selected: { type: Boolean, default: false },
    weekend: { type: Boolean, default: false },
    hidden: { type: Boolean, default: false },
    focused: { type: Boolean, default: false },
    highlightWeekend: { type: Boolean, default: false },
    highlightToday: { type: Boolean, default: false },
    readonly: { type: Boolean, default: false },
    color: { type: String, default: "primary" }
  },
  emits: ["click"],
  setup(props, { emit }) {
    const onClick = () => {
      if (!props.disabled) {
        emit("click");
      }
    };
    const { getColor } = useColors();
    const bg = computed(() => getColor(props.color));
    const { textColorComputed } = useTextColor(bg);
    return {
      bg,
      onClick,
      textColorComputed
    };
  }
});
const __injectCSSVars__ = () => {
  useCssVars((_ctx) => ({
    "4287396d": _ctx.bg,
    "6cf74d8a": _ctx.textColorComputed
  }));
};
const __setup__ = __default__.setup;
__default__.setup = __setup__ ? (props, ctx) => {
  __injectCSSVars__();
  return __setup__(props, ctx);
} : __injectCSSVars__;
const _sfc_main = __default__;
const _hoisted_1 = {
  key: 0,
  class: "va-date-picker-cell va-date-picker-cell_clear"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return _ctx.hidden ? (openBlock(), createElementBlock("div", _hoisted_1)) : (openBlock(), createElementBlock("div", {
    key: 1,
    class: normalizeClass(["va-date-picker-cell", {
      "va-date-picker-cell_other-month": _ctx.otherMonth,
      "va-date-picker-cell_today": _ctx.highlightToday && _ctx.today,
      "va-date-picker-cell_in-range": _ctx.inRange,
      "va-date-picker-cell_disabled": _ctx.disabled,
      "va-date-picker-cell_highlighted-weekend": _ctx.highlightWeekend && _ctx.weekend,
      "va-date-picker-cell_selected": _ctx.selected,
      "va-date-picker-cell_focused": _ctx.focused,
      "va-date-picker-cell_readonly": _ctx.readonly
    }]),
    onClick: _cache[0] || (_cache[0] = (...args) => _ctx.onClick && _ctx.onClick(...args)),
    onKeypress: _cache[1] || (_cache[1] = withKeys(withModifiers((...args) => _ctx.onClick && _ctx.onClick(...args), ["prevent", "stop"]), ["space", "enter"]))
  }, [
    renderSlot(_ctx.$slots, "default")
  ], 34));
}
var VaDatePickerCell = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { VaDatePickerCell as V };
