// vuestic-ui/dist/esm-node/src/components/va-card/components/va-card-block/VaCardBlock.mjs
import { defineComponent, computed, openBlock, createElementBlock, normalizeClass, renderSlot } from "vue";
import { _ as _export_sfc } from "../../../../../plugin-vue_export-helper.mjs";
var VaCardBlock_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ".va-card-block{display:flex;flex-direction:column}.va-card-block>div:first-child{border-top-right-radius:var(--va-card-border-radius);border-top-left-radius:var(--va-card-border-radius)}.va-card-block>div:last-child{border-bottom-right-radius:var(--va-card-border-radius);border-bottom-left-radius:var(--va-card-border-radius)}.va-card-block>div{flex:auto}.va-card-block--horizontal{flex-direction:row}.va-card-block--horizontal>div:first-child{border-top-left-radius:var(--va-card-border-radius);border-top-right-radius:0;border-bottom-right-radius:0;border-bottom-left-radius:var(--va-card-border-radius)}.va-card-block--horizontal>div:last-child{border-top-left-radius:0;border-bottom-left-radius:0;border-top-right-radius:var(--va-card-border-radius);border-bottom-right-radius:var(--va-card-border-radius)}.va-card-block--horizontal>.va-divider{flex:0}\n")();
const _sfc_main = defineComponent({
  name: "VaCardBlock",
  props: {
    horizontal: {
      type: Boolean,
      default: false
    }
  },
  setup: (props) => ({
    classComputed: computed(() => ({
      "va-card-block--horizontal": props.horizontal
    }))
  })
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", {
    class: normalizeClass(["va-card-block", _ctx.classComputed])
  }, [
    renderSlot(_ctx.$slots, "default")
  ], 2);
}
var VaCardBlockBase = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export { VaCardBlockBase as V };
