// vuestic-ui/dist/esm-node/src/components/va-timeline/VaTimelineItem/VaTimelineItem.mjs
import { defineComponent, h } from "vue";



import { u as useComponentPresetProp } from "../../../composables/useComponentPreset.mjs";



import { e as extractComponentProps } from "../../../utils/component-options/extract-component-options.mjs";
import { f as filterComponentProps } from "../../../utils/component-options/filter-props.mjs";
import { V as VaTimelineSeparator } from "../VaTimelineSeparator/index.mjs";
var VaTimelineItem_vue_vue_type_style_index_0_lang = /* @__PURE__ */ (() => ":root{--va-timeline-item-display: flex;--va-timeline-item-flex-direction: column;--va-timeline-item-vertical-flex-direction: row;--va-timeline-item-vertical-flex-wrap: nowrap;--va-timeline-item-vertical-align-items: stretch;--va-timeline-item-title-text-align: center;--va-timeline-item-title-color: var(--va-primary);--va-timeline-item-title-font-weight: 700;--va-timeline-item-title-font-size: .8rem;--va-timeline-item-title-text-transform: uppercase;--va-timeline-item-description-margin-top: .25rem;--va-timeline-item-description-text-align: center}.va-timeline-item{display:var(--va-timeline-item-display);flex-direction:var(--va-timeline-item-flex-direction)}.va-timeline-item__before,.va-timeline-item__after{flex:1}.va-timeline-item--vertical{flex-direction:var(--va-timeline-item-vertical-flex-direction);flex-wrap:var(--va-timeline-item-vertical-flex-wrap);align-items:var(--va-timeline-item-vertical-align-items)}.va-timeline-item--vertical .va-timeline-item__before,.va-timeline-item--vertical .va-timeline-item__after{padding-top:1rem;padding-bottom:1rem}.va-timeline-item--vertical .va-timeline-item__before{padding-right:1rem}.va-timeline-item--vertical .va-timeline-item__after{padding-left:1rem}.va-timeline-item:not(.va-timeline-item--vertical) .va-timeline-item__before,.va-timeline-item:not(.va-timeline-item--vertical) .va-timeline-item__after{padding-right:1rem;padding-left:1rem}.va-timeline-item:not(.va-timeline-item--vertical) .va-timeline-item__before{padding-bottom:.5rem}.va-timeline-item:not(.va-timeline-item--vertical) .va-timeline-item__after{padding-top:.5rem}\n")();
const COMPONENT_NAME = "va-timeline-item";
const VaTimelineSeparatorProps = extractComponentProps(VaTimelineSeparator);
const _sfc_main = defineComponent({
  name: COMPONENT_NAME,
  props: {
    ...useComponentPresetProp,
    ...VaTimelineSeparatorProps,
    color: { type: String, default: "primary" },
    isFirst: { type: Boolean },
    isLast: { type: Boolean },
    inverted: { type: Boolean }
  },
  setup(props, { slots }) {
    const children = [
      h(VaTimelineSeparator, { ...filterComponentProps(VaTimelineSeparatorProps).value })
    ];
    const before = props.inverted ? slots.after : slots.before;
    if (before) {
      children.unshift(h("div", { class: `${COMPONENT_NAME}__before` }, before()));
    }
    const after = props.inverted ? slots.before : slots.after;
    if (after) {
      children.push(h("div", { class: `${COMPONENT_NAME}__after` }, after()));
    }
    return () => h("div", {
      class: [
        { [COMPONENT_NAME]: true },
        { [`${COMPONENT_NAME}--vertical`]: props.vertical },
        { [`${COMPONENT_NAME}--is-first`]: props.isFirst },
        { [`${COMPONENT_NAME}--is-last`]: props.isLast }
      ]
    }, children);
  }
});
export { _sfc_main as _ };
