// vuestic-ui/dist/es/src/components/va-virtual-scroller/useVirtualScrollerSizes.js
import { shallowRef, computed, ref, getCurrentInstance, onMounted, watch } from "vue";





import { u as useEvent } from "../../composables/useEvent.js";

const isParsableMeasure = (value) => {
  if (typeof value === "string") {
    return !isNaN(+value) || value.endsWith("px") || value.endsWith("rem");
  }
  return false;
};
const isParsablePositiveMeasure = (value) => {
  if (typeof value === "number") {
    return value >= 0;
  }
  return isParsableMeasure(value) && parseInt(value) >= 0;
};
const validateSizeProp = (v, propName) => {
  const isProperValue = isParsablePositiveMeasure(v);
  return isProperValue;
};
const useVirtualScrollerSizesProps = {
  horizontal: { type: Boolean, default: false },
  itemSize: {
    type: [Number, String],
    default: 0,
    validator: (v) => {
      return validateSizeProp(v);
    }
  },
  wrapperSize: {
    type: [Number, String],
    default: 100,
    validator: (v) => {
      return validateSizeProp(v);
    }
  }
};
const useVirtualScrollerSizes = (props, scrollPosition) => {
  const list = shallowRef();
  const wrapper = shallowRef();
  const parseSizeValue = (value) => {
    if (typeof value === "string") {
      const parsedValue = parseInt(value);
      if (isNaN(parsedValue)) {
        return 0;
      }
      return value.endsWith("rem") ? parsedValue * pageFontSize.value : parsedValue;
    }
    return value;
  };
  const wrapperSize = computed(() => {
    return parseSizeValue(props.wrapperSize);
  });
  const pageFontSize = ref(16);
  const handleWindowResize = () => {
    pageFontSize.value = parseFloat(getComputedStyle(document.documentElement).fontSize);
  };
  useEvent("resize", handleWindowResize, true);
  const itemSizeCalculated = ref(0);
  const calcAverageItemsSize = () => {
    if (!list.value) {
      return;
    }
    const sizes = [];
    const itemsList = list.value.children;
    const itemsAmount = itemsList.length;
    for (let i = 0; i < itemsAmount; i++) {
      const currentChild = list.value.children.item(i);
      currentChild && sizes.push(currentChild[props.horizontal ? "clientWidth" : "clientHeight"]);
    }
    itemSizeCalculated.value = itemsAmount ? Math.trunc(sizes.reduce((acc, el) => acc + el, 0) / (itemsAmount - 1)) : 0;
  };
  const instance = getCurrentInstance();
  onMounted(() => {
    var _a, _b;
    if (!list.value) {
      list.value = (_b = (_a = instance == null ? void 0 : instance.parent) == null ? void 0 : _a.refs) == null ? void 0 : _b.list;
    }
    calcAverageItemsSize();
  });
  watch(scrollPosition, calcAverageItemsSize);
  watch(wrapperSize, calcAverageItemsSize);
  let oldItemSize = 0;
  const itemSize = computed(() => {
    const sizeParsed = parseSizeValue(props.itemSize);
    const result = Math.max(sizeParsed, itemSizeCalculated.value, 1);
    const diff = Math.abs(oldItemSize / result * 100 - 100);
    if (diff > 5 || oldItemSize === 0) {
      oldItemSize = result;
      return result;
    }
    return oldItemSize;
  });
  return { list, wrapper, itemSize, wrapperSize };
};
export { useVirtualScrollerSizes as a, useVirtualScrollerSizesProps as u };
